torchgeo/trainers/detection.py

```diff
@@ -151,6 +151,10 @@ def configure_models(self) -> None:
                 num_classes,
                 rpn_anchor_generator=anchor_generator,
                 box_roi_pool=roi_pooler,
+                min_size=800,
+                max_size=800,
```

Comment:
Is this a hard constraint? It would be nice to be as flexible as possible here.

Reply:
I actually think it might be better to pass this in as an arg, image_size or similar. However as mentioned it would be preferable if we could avoid this processing altogether.

Reply:
I'll make it an input arg

Reply:
Do all the datamodules support image_size as an input arg? I have it as one for mine and always perform a resize - definitely want control over this

Reply:
Not consistently, no, but we could. We definitely could for all the object detection/instance segmentation ones, there aren't that many.

Reply:
Let me try updating the datamodules with controllable sizes and see if I can get the models to converge. Ideally we could set this range to be a no-op which is what I originally set it to (1,4096) for